### PR TITLE
Molpath PHCP

### PR DESCRIPTION
**Pull request by jiaan-yu**
Main update:
- molpath tumor only pipeline (PHCP), based on the structure of FRCP in bitbucket.
- tumor only variant calling (Mutect2)

Other introduced changes:
- new subworkflow GATKBaseRecalBam: BaseRecalibrator + ApplyBQSR
- GATK variant calling section will be break into: GATKBaseRecalBam + Variant Calling + (Combine Variants) + Bamstats. This has done for Mutect2, and will break down the HaplotypeCaller

Minor changes:
Did you get error when running 
janis_bioinformatics/tools/vcflib/vcflength/base.py?
Parsing vcf.gz needs tbi index as well.

### CODE REVIEW

**Comment by jiaan-yu** [comment]
Minor change to
janis_bioinformatics/tools/pmac/molpathGermlineWorkflow.py
 - Reconstruct the subworkflows for HaplotypeCaller variant calling to match with the Mutect2
 - Fixes for bed files

File: janis_bioinformatics/tools/common/gatkbasecalbam.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from janis_bioinformatics.data_types import BamBai, FastaWithDict, VcfTabix, Bed
+from janis_bioinformatics.tools.gatk4 import (
+    Gatk4BaseRecalibrator_4_1_3,
+    Gatk4ApplyBqsr_4_1_3,
+)
+from janis_bioinformatics.tools.bioinformaticstoolbase import BioinformaticsWorkflow
+
+
+class GATKBaseRecalBam_4_1_3(BioinformaticsWorkflow):
```
Comment (illusional) [inline janis_bioinformatics/tools/common/gatkbasecalbam.py:9]: Per internal conversation, can we give this one a name that better describes what it does.

File: janis_bioinformatics/tools/common/uncompressvcf.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from janis_bioinformatics.data_types import Vcf, VcfTabix
+from janis_bioinformatics.data_types import Vcf, VcfTabix, File
```
[the remark this answers is not in the document]
Comment (jiaan-yu) [reply janis_bioinformatics/tools/common/uncompressvcf.py:1]: Tried this but doesn't work, any suggestions?
```
    def inputs(self) -> List[ToolInput]:
        return [
            ToolInput("file", Vcf(), position=100, doc="File to bgzip compress"),
            ToolInput(
                "outputFilename",
                Filename(extension=".vcf.gz"),
                position=102,
                shell_quote=False,
            ),
            *self.additional_args,
        ]

    def arguments(self):
        return [ToolArgument(">", position=101, shell_quote=False)]

    def outputs(self) -> List[ToolOutput]:
        return [
            ToolOutput("out", CompressedVcf(), glob=InputSelector("outputFilename"),)
        ]
```

File: janis_bioinformatics/tools/htslib/bgzip/base.py [hunk 3]
```diff
@@ -22,15 +30,19 @@ def base_command(self):
     def inputs(self) -> List[ToolInput]:
         return [
             ToolInput("file", Vcf(), position=100, doc="File to bgzip compress"),
+            ToolInput(
+                "outputFilename",
+                Filename(extension=".vcf.gz"),
+                prefix=">",
```
[the remark this answers is not in the document]
Comment (jiaan-yu) [reply janis_bioinformatics/tools/htslib/bgzip/base.py:36]: Tried this but didn't work, any suggestion?
```
    def inputs(self) -> List[ToolInput]:
        return [
            ToolInput("file", Vcf(), position=100, doc="File to bgzip compress"),
            ToolInput(
                "outputFilename",
                Filename(extension=".vcf.gz"),
                position=102,
                shell_quote=False,
            ),
            *self.additional_args,
        ]

    def arguments(self):
        return [ToolArgument(">", position=101, shell_quote=False)]

    def outputs(self) -> List[ToolOutput]:
        return [
            ToolOutput("out", CompressedVcf(), glob=InputSelector("outputFilename"),)
        ]
```

File: janis_bioinformatics/tools/common/gatkbasecalbam.py (hunk 1 above)
Comment (jiaan-yu) [reply janis_bioinformatics/tools/common/gatkbasecalbam.py]: fix in commit 
4633b67

File: janis_bioinformatics/tools/common/uncompressvcf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jiaan-yu) [reply janis_bioinformatics/tools/common/uncompressvcf.py]: fix in 
ee1a658
Also check janis-unix for PR

File: janis_bioinformatics/tools/htslib/bgzip/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jiaan-yu) [reply janis_bioinformatics/tools/htslib/bgzip/base.py]: 08f021c

**Comment by illusional** [comment]
Thanks @jiaan-yu, merged!

**Pull request merged**